tiledb/ml/readers/tensorflow_sparse.py

```diff
@@ -200,12 +242,17 @@ def _generator_sparse_dense(
             x_coords[0] -= x_coords[0].min()
 
             # for the check slice the row dimension of y dense array
-            cls.__check_row_dims(x_coords[0], values_y, sparse=False)
-
-            x_data = np.array(values_x).flatten()
+            cls.__check_row_dims(
+                x_coords[0], y_batch[y_attribute_names[0]], sparse=False
+            )
 
-            yield tf.sparse.SparseTensor(
-                indices=constant_op.constant(list(zip(*x_coords)), dtype=tf.int64),
-                values=constant_op.constant(x_data, dtype=x_dtype),
-                dense_shape=(batch_size, x_shape[0]),
-            ), values_y
+            yield tuple(
+                tf.sparse.SparseTensor(
+                    indices=constant_op.constant(list(zip(*x_coords)), dtype=tf.int64),
+                    values=constant_op.constant(
+                        x_batch[attr].flatten(), dtype=x.schema.attr(attr).dtype
+                    ),
+                    dense_shape=(batch_size, x_shape[0]),
+                )
+                for attr in x_attribute_names
+            ) + tuple(y_batch[attr] for attr in y_attribute_names)
```

Comment:
Why adding the `x` and `y` tuples here? Don't we want them separated?

Reply:
They are concatenated. It's exactly the same as  `return attr_1, attr_2, ...`